src/i2c.rs

```diff
@@ -246,6 +249,62 @@ where
                 token,
                 channel,
                 buffer,
+                num_words,
+                address,
+                dma::Priority::high(),
+                dma::Direction::memory_to_peripheral(),
+            )
+        };
+
+        Transfer {
+            target: self,
+            inner:  transfer,
+        }
+    }
+
+    #[cfg(feature = "stm32l0x2")]
+    pub fn write_some<Channel, Buffer>(mut self,
+        dma:     &mut dma::Handle,
+        channel: Channel,
+        address: u8,
+        buffer:  Pin<Buffer>,
+        num_words: usize,
+    )
+        -> Transfer<Self, Tx<I>, Channel, Buffer, dma::Ready>
+        where
+            Tx<I>:          dma::Target<Channel>,
+            Channel:        dma::Channel,
+            Buffer:         Deref + 'static,
+            Buffer::Target: AsSlice<Element=u8>,
+    {
+        assert!(buffer.len() >= num_words);
+        self.start_transfer(address, buffer.as_slice().len(), RD_WRNW::WRITE);
+
+        // This token represents the transmission capability of I2C and this is
+        // what the `dma::Target` trait is implemented for. It can't be
+        // implemented for `I2c` itself, as that would allow for the user to
+        // pass, for example, a channel that can do I2C RX to `write_all`.
+        //
+        // Theoretically, one could create both `Rx` and `Tx` at the same time,
+        // or create multiple tokens of the same type, and use that to create
+        // multiple simultaneous DMA transfers, which would be wrong and is not
+        // supported by the I2C peripheral. We prevent that by only ever
+        // creating an `Rx` or `Tx` token while we have ownership of `I2c`, and
+        // dropping the token before returning ownership of `I2c` ot the user.
+        let token = Tx(PhantomData);
+
+        // Safe, because we're only taking the address of a register.
+        let address = &unsafe { &*I::ptr() }.txdr as *const _ as u32;
+
+        // Safe, because the trait bounds of this method guarantee that the
+        // buffer can be read from.
+        let transfer = unsafe {
+            dma::Transfer::new(
+                dma,
+                token,
+                channel,
+                buffer,
+                num_words,
```

Comment:
It looks like you just copy-pasted `write_all`. Can't you implement `write_all` by calling `write_some`, to remove the duplicated code?

Reply:
Of course, this was pretty lazy on my part.  Fixed for all cases.